### DISCUSSION TITLE
How to set the maximum number of CPU cores for the Dpgen main program?

### DISCUSSION THREAD

**Discussion by h840473807** [deepmodeling/dpgen#1205] [Q&A] [answered]
I am running Dpgen on a login node of a supercomputer cluster. When generating “02.fp", it automatically uses all available cores, causing the login node to become slow. Is it possible to set the maximum number of CPU cores to be used, similar to "make -j n"?
Thank you

**Comment by njzjz** [comment, marked as answer]
DP-GEN uses NumPy to process data so you can refer to [NumPy documentation](https://numpy.org/devdocs/reference/global_state.html).

Note: We do not recommend any program running on the login node of the cluster.

**Reply by h840473807** [reply]
Thank you, I have set variables "export OMP_NUM_THREADS = 4" and solved the problem.
